components/sup-protocol/src/lib.rs

```diff
@@ -168,3 +181,52 @@ pub fn socket_addr_env_or_default(env_var: &str, default: SocketAddr) -> SocketA
         .parse()
         .unwrap_or(default)
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::fs::File;
```

Comment:
With the above suggestions, you only need to add the one import here to cover the use of the `write!` macro.
```suggestion
    use std::fs::File;
    use std::io::Write
```

Reply:
completely agreed.